WordPress/openverse
# Refactor the error page

## Fixes



Related to #632 by @sarayourfriend

## Description


_Note: this PR was split from #2838 to make the PR review easier. ~~More than 40 changed files here are the deleted snapshots~~ Snapshot changes were extracted to a separate PR_

This PR adds a new `VErrorPage` component that can show the client-side error using only the `fetchState.fetchingError` as a prop. This is simpler than the previous version that required providing the image and the error code and other necessary data separately. This will allow us in the future to add details to the error page easily. For instance, we will be able to show how much time is left until next retry on  429 error.

The layouts were updated a little bit based on the [i3 mockups](https://www.figma.com/file/H1qjjlII3PaDltPqW59Vj7/No-results-page?type=design&node-id=1001-6695&mode=dev). Those are not finalized, but I think they are the most up-to-date version that we can use until the new designs are ready.

~~The `VSearchGrid` component was also removed because it overcomplicates the layout, requires you to pass too many props, and uses a slot for `media`(audio/image) search page that breaks the context. Now `search` page handles the display of the search header above the media results and the external search form below them. It also shows the error page instead of the results when there is an error.~~ - this was moved to #2866

~~The visual regression tests were refactored: only "mobile" and "desktop" variants are checked (instead of all breakpoints) to save on the snapshot files. I believe that this is enough for testing that the error pages are displayed correctly. 
All of the error page tests were moved to `errors.spec.ts`, the names were standardized, and if some error pages are the same, the same snapshot was reused (e.g., 500, 429 use the same "generic" error, at least before we implement the new error page designs).~~

## Testing Instructions

The main changes of this PR are in the `VSearchPage.vue` and `search.vue` components, and the `VErrorPage` stories. These changes should make sense.

To test the error page, try searching for something that does not have any results ("querywithnoresults") - you should see the No results error page. 

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
